### PR TITLE
feat: allow non-reputable users to link

### PR DESCRIPTION
**Pull request by clD11**
### Summary

Allow non-reputable users to link
* remove unused parameters

### Type of change ( select one )

- [x] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: services/wallet/datastore.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func init() {
 // Datastore holds the interface for the wallet datastore
 type Datastore interface {
 	datastore.Datastore
-	LinkWallet(ctx context.Context, ID string, providerID string, providerLinkingID uuid.UUID, anonymousAddress *uuid.UUID, depositProvider, country string) error
+	LinkWallet(ctx context.Context, ID string, providerID string, providerLinkingID uuid.UUID, depositProvider string) error
```
Comment (pavelbrm) [inline services/wallet/datastore.go:74]: Nit: While here, we could tidy it up further:

```suggestion
	LinkWallet(ctx context.Context, id, providerID string, providerLinkingID uuid.UUID, depositProvider string) error
```

File: services/wallet/datastore.go [hunk 2]
```diff
@@ -561,7 +561,7 @@ var (
 )
 
 // LinkWallet links a wallet together
-func (pg *Postgres) LinkWallet(ctx context.Context, ID string, userDepositDestination string, providerLinkingID uuid.UUID, anonymousAddress *uuid.UUID, depositProvider, country string) error {
+func (pg *Postgres) LinkWallet(ctx context.Context, ID string, userDepositDestination string, providerLinkingID uuid.UUID, depositProvider string) error {
```
Comment (pavelbrm) [inline services/wallet/datastore.go:564]: Similar nitpick as above, but would require more unrelated changes – the code uses `ID` and then later creates `id` from it. 🤦‍♂️

**Review by pavelbrm — APPROVED** [review]
Approving with non-blocking nitpicks. 👍

File: services/wallet/datastore.go (hunk 2 above)
Comment (clD11) [reply services/wallet/datastore.go:564]: added to the wallet refactor ticket

File: services/wallet/datastore.go (hunk 1 above)
Comment (clD11) [reply services/wallet/datastore.go:74]: added to the wallet refactor ticket

**Pull request merged**